### PR TITLE
feat: add seed to reproduce simulations

### PR DESCRIPTION
**Pull request by pablobd**
In this PR, I add the possibility to set a seed, so that simulations are reproducible. Also it will allow to make more fair comparisons betweens methods of analysis.

The main sources of stochasticity that are controlled with the seed are:

1. The splitter
2. The binary perturbator

### CODE REVIEW

File: cluster_experiments/power_analysis.py [hunk 1]
```diff
@@ -115,6 +119,10 @@ def __init__(
             target_col=target_col,
             features_cupac_model=features_cupac_model,
         )
+        if seed:
```
Comment (david26694) [inline cluster_experiments/power_analysis.py:122]:
```suggestion
        if seed is not None:
```

perhaps an edge case, but if seed = 0, won't seed anything

File: tests/power_analysis/test_seed.py [hunk 2]
```diff
@@ -0,0 +1,37 @@
+import numpy as np
+
+from cluster_experiments.perturbator import Perturbator
+from cluster_experiments.power_analysis import PowerAnalysis
+
+
+def get_config(perturbator: Perturbator) -> dict:
```
Comment (david26694) [inline tests/power_analysis/test_seed.py:7]:
```suggestion
def get_config(perturbator: str) -> dict:
```

I think it's a string

**Review by david26694 — COMMENTED** [review]
couple of minor comments

File: cluster_experiments/power_analysis.py (hunk 1 above)
Comment (pablobd) [reply cluster_experiments/power_analysis.py]: Good catch! I changed it to your suggestion

File: tests/power_analysis/test_seed.py (hunk 2 above)
Comment (pablobd) [reply tests/power_analysis/test_seed.py]: True, changed it too

**Review by david26694 — APPROVED** [review]
lgtm!

**Review by alexanderhensel — APPROVED** [review]
LGTM
